src/geode/model/helpers/component_mesh_queries.cpp

```diff
@@ -109,10 +112,87 @@ namespace
         }
         return {};
     }
+
+    absl::optional< geode::PolygonVertices >
+        block_unique_facet_vertices_from_surface_polygon(
+            const geode::BRep& model,
+            const geode::Block3D& block,
+            const geode::PolygonVertices& polygon_unique_vertices )
+    {
+        geode::PolygonVertices block_facet_vertices;
+        for( const auto polygon_vertex :
+            geode::LIndices{ polygon_unique_vertices } )
+        {
+            const auto block_vertex_indices = model.mesh_component_vertices(
+                polygon_unique_vertices[polygon_vertex], block.id() );
+            if( block_vertex_indices.size() == 1 )
+            {
+                block_facet_vertices.emplace_back( block_vertex_indices[0] );
+            }
+            else
+            {
+                return absl::nullopt;
```

Comment:
So it does no return something if surface is internal?

Reply:
If the surface is internal or if the a polygon vertex has several mesh_component_vertices associated, it returns a null_opt (e.g. it cannot find the vertices from this information only).